lib/ruby_lsp/listener.rb

```diff
@@ -2,13 +2,18 @@
 # frozen_string_literal: true
 
 module RubyLsp
+  # Listener is an abstract class to be used by requests for listening to events emitted when visiting an AST using the
+  # EventEmitter.
   class Listener
     extend T::Sig
     extend T::Helpers
     include Requests::Support::Common
 
     abstract!
 
+    sig { abstract.returns(Object) }
```

Comment:
It may be useful to document at least the abstract methods so implementers know how they should override them. An example would be nice too.

Reply:
Is there any kind of common interface we could use to type the return value? If not should we be introducing one? Or how is the system going to handle every kind of response here?